### PR TITLE
Ra 2020

### PR DESCRIPTION
**Pull request by fawques**

### CODE REVIEW

File: LCU.NET.Extras/App.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using Ninject;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LCU.NET.Extras
+{
+    public static class LCUApp
```
Comment (fawques) [inline LCU.NET.Extras/App.cs:11]: I don't really like this structure of a static class, Is there a better way?

File: LCU.NET.Extras/Game/Detector.cs [hunk 2]
```diff
@@ -15,12 +15,14 @@ public abstract class Detector
     {
         protected Container<Actuator.State> State { get; private set; }
         protected ILoL LoL { get; }
+        protected IUiActuator MainWindow { get; }
```
Comment (fawques) [inline LCU.NET.Extras/Game/Detector.cs:18]: I don't really like the naming. I'ts like that to keep the original names, but is there a better way?

File: LCU.NET.Extras/App.cs (hunk 1 above)
Comment (VoldarGames) [reply LCU.NET.Extras/App.cs:11]: Maybe this class can be deleted and move those properties to the application service

File: LCU.NET.Extras/Data/Config.cs [hunk 3]
```diff
@@ -103,8 +103,8 @@ public void Save()
             }
             catch
             {
-                if (!DesignerProperties.GetIsInDesignMode(new DependencyObject()))
-                    throw;
+                //if (!DesignerProperties.GetIsInDesignMode(new DependencyObject()))
```
Comment (VoldarGames) [inline LCU.NET.Extras/Data/Config.cs]: if you dont use something, delete it

File: LCU.NET.Extras/Data/Config.cs [hunk 4]
```diff
@@ -121,7 +121,7 @@ private static Config Load()
 
             try
             {
-                if (!File.Exists(FilePath) || MainWindow.InDesigner)
+                if (!File.Exists(FilePath)/* || MainWindow.InDesigner*/)
```
Comment (VoldarGames) [inline LCU.NET.Extras/Data/Config.cs]: Same as above

File: LCU.NET.Extras/Data/NotificationType.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LCU.NET.Extras.Data
+{
+    // Copy of Notifications.Wpf.NotificationType to avoid references to UI.
+    public enum NotificationType
```
Comment (VoldarGames) [inline LCU.NET.Extras/Data/NotificationType.cs:10]: Are you using this enum?

File: LCU.NET.Extras/Data/Riot.cs [hunk 6]
```diff
@@ -208,7 +207,7 @@ public static async Task CacheAllAsync(Action<double> progress)
             int p = 0;
             await Task.WhenAll(total.Select(async o =>
             {
-                await ImageCache.Instance.Get(o);
+                //await ImageCache.Instance.Get(o);
```
Comment (VoldarGames) [inline LCU.NET.Extras/Data/Riot.cs]: ?

File: LCU.NET.Extras/Game/Actuator.Game.cs [hunk 7]
```diff
@@ -234,7 +233,7 @@ public async Task UploadRunePage(Position pos, int championId)
 
             LogTo.Debug("for champion {0}", champion);
 
-            Main.ShowNotification(Text.LockedInMessage, champion + ", " + pos.ToString().ToLower(), NotificationType.Success);
+            Main.ShowNotification("Locked in", champion + ", " + pos.ToString().ToLower(), NotificationType.Success);
```
Comment (VoldarGames) [inline LCU.NET.Extras/Game/Actuator.Game.cs:236]: I don't like those hardcoded strings but... i don't know how to solve this without a dependency to our system

File: LCU.NET.Extras/LCU.NET.Extras.csproj.nuspec [hunk 8]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<package >
+  <metadata>
+    <id>LCU.NET.Extras</id>
+    <version>1.0.0</version>
+    <authors>Fluendo S.A.</authors>
```
Comment (VoldarGames) [inline LCU.NET.Extras/LCU.NET.Extras.csproj.nuspec]: Remind to give credit to the real author, put his license, etc..

File: Legendary Rune Maker/Pages/MainPage.xaml.cs [hunk 9]
```diff
@@ -379,10 +381,10 @@ private async void Page_Initialized(object sender, EventArgs e)
             await SetChampion(null);
         }
 
-        public void ShowNotification(string title, string message = null, NotificationType type = NotificationType.Information)
+        public void ShowNotification(string title, string message = null, LCU.NET.Extras.Data.NotificationType type = LCU.NET.Extras.Data.NotificationType.Information)
```
Comment (VoldarGames) [inline Legendary Rune Maker/Pages/MainPage.xaml.cs:384]: review using statements on whole file

**Review by VoldarGames — CHANGES_REQUESTED** [review]

File: LCU.NET.Extras/App.cs (hunk 1 above)
Comment (fawques) [reply LCU.NET.Extras/App.cs:11]: I need to access these properties from this project (LCU.NET.Extras), but I'm not adding a dependency with Rift.Services here.

File: LCU.NET.Extras/Data/NotificationType.cs (hunk 5 above)
Comment (fawques) [reply LCU.NET.Extras/Data/NotificationType.cs:10]: The code here is using it, in the end in our implementation we always send Info, but we probably will change it later.

File: LCU.NET.Extras/Game/Actuator.Game.cs (hunk 7 above)
Comment (fawques) [reply LCU.NET.Extras/Game/Actuator.Game.cs:236]: Yep, I was thinking of just removing the notifications, or adding i18n here. Talking with @axelgorris, he managed it in his story, and luna agrees that it's left as only english.

File: Legendary Rune Maker/Pages/MainPage.xaml.cs (hunk 9 above)
Comment (fawques) [reply Legendary Rune Maker/Pages/MainPage.xaml.cs:384]: I'm not really mantaining the MainPage of the Legendary Rune Maker, so... ¯\\\_(ツ)_/¯

**Review by VoldarGames — APPROVED** [review]

**Pull request merged**